lib/src/acter_avatar.dart

```diff
@@ -115,32 +92,18 @@ class _ActerAvatar extends State<ActerAvatar> {
 
   @override
   Widget build(BuildContext context) {
+    final avatarInfo = widget.options.avatar;
     final child =
         GestureDetector(onTap: widget.onAvatarTap, child: inner(context));
-    switch (widget.tooltip) {
-      case TooltipStyle.DisplayName:
-        return Tooltip(
-          message: widget.avatarInfo.displayName ?? widget.avatarInfo.uniqueId,
-          child: child,
-        );
-      case TooltipStyle.UniqueId:
-        return Tooltip(message: widget.avatarInfo.uniqueId, child: child);
-      case TooltipStyle.Combined:
-        var message =
-            widget.avatarInfo.uniqueName ?? widget.avatarInfo.uniqueId;
-        if (widget.avatarInfo.displayName != null) {
-          message =
-              '${widget.avatarInfo.displayName} (${widget.avatarInfo.uniqueId})';
-        }
-        return Tooltip(message: message, child: child);
-      case TooltipStyle.None:
-        return child;
-    }
+    return tooltipMsgWithChild(avatarInfo, child);
```

Comment:
This has also been reduced to have one singular widget method, both for main avatar and parent badges tooltip display.